server/api/controllers/reports.js

```diff
@@ -0,0 +1,37 @@
+import { ArticleReports, Article } from '../../models';
+
+/**
+ *
+ * @function reportArticle
+ * @param {*} req any
+ * @param {*} res  any
+ * @returns {Object} JSON object
+ */
+export const reportArticle = async (req, res) => {
+  try {
+    const userId = req.user.id;
+    const articleId = req.params.id;
+    const newReport = {
+      userId,
+      articleId
+    };
+    const article = await Article.findById(articleId);
+    if (!article) {
+      throw new Error('Article is not found');
```

Comment:
@Princesso What's your thought behind using the `throw new Error` syntax ? What do u intend to achieve ? Please share.

Reply:
In the code I had written previously, there were nested if statements, and testing each behaviour was difficult. I went to research on how to avoid that. I found out that ensuring the program flow does not continue if a condition is false is better that nesting it. The throw Error syntax also returns a custom message that is passed into it. I have tested the behaviour of the throw Error method and it returns the custom message I passed into it.